File: libnetwork/src/main/java/com/gystry/libnetwork/cache/DateConvert.java
package com.gystry.libnetwork.cache;

import androidx.room.TypeConverter;

import java.util.Date;

/**
 * @author gystry
 * 创建日期：2020/7/7 21
 * 邮箱：gystry@163.com
 * 描述：
 */
public class DateConvert {
    @TypeConverter
    public static Long date2Long(Date date){
        return date.getTime();
    }

    @TypeConverter
    public static Date long2Date(Long date){
        return new Date(date);
    }
}

File: libnetworkkt/src/main/java/com/gystry/libnetworkkt/cache/DateConvert.kt
package com.gystry.libnetworkkt.cache

import androidx.room.TypeConverter
import java.util.*

/**
 * @author gystry
 * 创建日期：2021/3/18 18
 * 邮箱：gystry@163.com
 * 描述：
 */
object DateConvert {
    @TypeConverter
    fun date2Long(date: Date):Long=date.time

    @TypeConverter
    fun long2Date(date:Long):Date= Date(date)
}